test/plugins/fingerprint.pulgin.spec.ts

```diff
@@ -0,0 +1,102 @@
+import { VerifyFiltersDto } from '../../src/plugins/dto/verify.filters.dto';
+import { FingerprintPlugin } from '../../src/plugins/impl/fingerprint.plugin';
+import { MockBioAuthService } from '../mock/mock.bio.auth.service';
+import { ExternalId } from '../../src/db/entity/external.id';
+import { MockRepository } from '../mock/mock.repository';
+import { now, pepperHash } from '../support/functions';
+import { VerifyFingerprintImageDto } from '../../src/plugins/dto/verify.fingerprint.image.dto';
+import { ProtocolException } from 'protocol-common/protocol.exception';
+import { ProtocolErrorCode } from 'protocol-common/protocol.errorcode';
+
+export class MockFingerpintPlugin implements VerifyFiltersDto {
+
+    constructor(private readonly status: string, private readonly id: string, private readonly agentId: string) {}
+
+    externalIds: object;
+    async getIds(filters: VerifyFiltersDto): Promise<Map<string, string>> {
+        return new Map<string, string>(Object.entries(filters.externalIds));
+    }
+
+    async verify(params: any, filters: VerifyFiltersDto): Promise<{}> {
+        return Promise.resolve({
+            data: {
+                status: this.status,
+                did: this.id,
+            },
+        });
+    }
+
+    async verifyFingerprint(params: any, position: number, image: string, agentId: string): Promise<any> {
+        return Promise.resolve({
+            data: {
+                status: this.status,
+                agentId: this.agentId,
+            },
+        });
+    }
+}
+describe('Fingerprint Plugin is valid', () => {
+    //const fingerprintVerifySpec = new VerifyFiltersDto ;
+    it('Plugin is not empty', async () => {
+        const agentId = 'agentId123';
+        const bioAuthService = new MockBioAuthService('not_matched', agentId);
+
+        const id2 = 1000000 + parseInt(now().toString().substr(7,6), 10);
+        const id1 = 'N' + id2;
+        const id1Hash = pepperHash(id1);
```

Comment:
This is a lot of work to create something that could just be any ol' string. I'd recommend deleting all of these lines and setting `mockExternalId.external_id = "foobar";` or something like that.